src/WebJobs.Script/DependencyInjection/ScriptStartupTypeLocator.cs

```diff
@@ -132,6 +132,13 @@ public async Task<IEnumerable<Type>> GetExtensionsStartupTypesAsync()
             }
             else
             {
+                string systemPath = Path.Combine(_rootScriptPath, ScriptConstants.AzureFunctionsSystemDirectoryName);
+
+                if (!Directory.Exists(systemPath) && isDotnetIsolatedApp)
```

Comment:
Consider checking isDotnetIsolatedApp first, if it's false, there is no need to compute the path or check if the directory exists